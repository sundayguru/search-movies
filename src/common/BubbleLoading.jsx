import React from 'react';
import BubblePreloader from 'react-bubble-preloader';


const BubbleLoading = (props) => {
    return (
      <div style={props.style} className="loading">
        {(props.loading && <BubblePreloader
            bubble={{ width: '1rem', height: '1rem' }}
            animation={{ speed: 1 }}
            colors={['#0167c2', '#00B5FF', '#DCEFFA']}
        />) || props.label}
        {props.loading && props.loadingText && <p style={{marginTop: 20}}>{props.loadingText}</p>}
      </div>
    );
};

export default BubbleLoading;