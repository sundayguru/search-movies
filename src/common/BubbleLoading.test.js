import React from 'react'
import BubbleLoading from './BubbleLoading'
import renderer from 'react-test-renderer'


test('<SearchBox />', () => {
  const component = renderer.create(
    <BubbleLoading loading={true} />,
  );

  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()

  const componentNotLoading = renderer.create(
    <BubbleLoading loading={false} />,
  );
  let treeNotLoading = componentNotLoading.toJSON()
  expect(treeNotLoading).toMatchSnapshot()
});
