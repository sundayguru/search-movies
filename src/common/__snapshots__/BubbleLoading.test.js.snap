// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBox /> 1`] = `
<div
  className="loading"
>
  <div
    className="bubble-loader"
  >
    <div
      className="bubble1"
      style={
        Object {
          "animation": "bouncedelay 1s ease-in-out infinite both",
          "animationDelay": "-0.32s",
          "backgroundColor": "#0167c2",
          "borderRadius": "50%",
          "display": "inline-block",
          "height": "1rem",
          "width": "1rem",
        }
      }
    />
    <div
      className="bubble2"
      style={
        Object {
          "animation": "bouncedelay 1s ease-in-out infinite both",
          "animationDelay": "-0.16s",
          "backgroundColor": "#00B5FF",
          "borderRadius": "50%",
          "display": "inline-block",
          "height": "1rem",
          "width": "1rem",
        }
      }
    />
    <div
      className="bubble3"
      style={
        Object {
          "animation": "bouncedelay 1s ease-in-out infinite both",
          "backgroundColor": "#DCEFFA",
          "borderRadius": "50%",
          "display": "inline-block",
          "height": "1rem",
          "width": "1rem",
        }
      }
    />
  </div>
</div>
`;

exports[`<SearchBox /> 2`] = `
<div
  className="loading"
/>
`;
